test/DynamoCoreTests/PythonMigrationAssistantTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Dynamo.PythonMigration.MigrationAssistant;
+using NUnit.Framework;
+using System.Collections.Generic;
+
+namespace Dynamo.Tests
+{
+    public class PythonMigrationAssistantTests
+    {
+        [Test]
+        public void CanMigratePyton2CodeToPython3Code()
+        {
+            // Arrange
+            var expectedPython3Code = @"print(""Hello, Dynamo!"")";
+
+            var python2Code = @"print ""Hello, Dynamo!""";
+
+            // Act
+            var migratedScript = ScriptMigrator.MigrateCode(python2Code);
+
+            // Assert
+            Assert.AreNotEqual(python2Code, migratedScript);
+            Assert.AreEqual(expectedPython3Code, migratedScript);
+        }
+
+        [Test]
+        public void MigrationWillNotChangePython3CompatibleCode()
+        {
+            // Arrange
+            var expectedPython3Code = @"for x in range(1, 5): print(x)";
+
+            // Act
+            var migratedScript = ScriptMigrator.MigrateCode(expectedPython3Code);
+
+            // Assert
+            Assert.AreEqual(expectedPython3Code, migratedScript);
+        }
+
+        [Test]
+        public void MigrationWillNotChangeCodeAfterBeingConverted()
+        {
+            // Arrange
+            var originalPython2Code = @"for x in xrange(100): continue";
+            var expectedPython3CodeAfterMigration = @"for x in range(100): continue";
+
+            // Act
+            var migratedScript = ScriptMigrator.MigrateCode(originalPython2Code);
+            var migratedScript2 = ScriptMigrator.MigrateCode(migratedScript);
+
+            // Assert
+            Assert.AreEqual(expectedPython3CodeAfterMigration, migratedScript);
+            Assert.AreEqual(expectedPython3CodeAfterMigration, migratedScript2);
+        }
+    }
```

Comment:
I don't see any tests regarding special cases of `migrate_2to3.py` like the `bad input` retry or the use of a `fixer`. If unit tests just can't be added, it would be nice to at least count with better code documentation about them. For instance, where would we place a fixer if we wanted to use one?

Reply:
@mmisol regarding the `fixers` as it is now im using the default fixers that ships with `lib2to3`, so adding to them might be complicated this way. We could ship our own `fixers` with Dynamo in some central location so that adding new fixers would just be a case of adding them to that location. Not sure what your thoughts are on that?

I think that the default fixers should handle most cases.

Reply:
Thanks for the feedback @SHKnudsen . I don't think we have a story for it, but at some point we discussed the possibility to address some `Iron Python => Native Python` compatibility issues using a custom fixer. My feeling is that we probably don't require to have that ability as part of this PR, and could do something like what you suggest later if required. What do you think about this @Amoursol @mjkkirschner ?

Reply:
I'll defer to @mjkkirschner on this one, but I agree @mmisol that we probably don't require it as part of this PR.

Reply:
Brought this to standup the other day and I didn't get any remarks from Mike. Unless he comments otherwise, let's assume we don't need this for now, for the sake of unblocking this PR.